packages/utils/src/path.ts

```diff
@@ -51,7 +51,7 @@ function normalizeArray(parts: string[], allowAboveRoot?: boolean): string[] {
 
 // Split a filename into [root, dir, basename, ext], unix version
 // 'root' is just a slash, or nothing.
-const splitPathRe = /^(\/?|)([\s\S]*?)((?:\.{1,2}|[^/]+?|)(\.[^./]*|))(?:[/]*)$/;
+const splitPathRe = /^(\S+:\\|\/?)([\s\S]*?)((?:\.{1,2}|[^/\\]+?|)(\.[^./\\]*|))(?:[/\\]*)$/;
```

Comment:
```suggestion
const splitPathRe = /^(\S+:\\|\/?|)([\s\S]*?)((?:\.{1,2}|[^/\\]+?|)(\.[^./\\]*|))(?:[/\\]*)$/;
```

Don't we need to do this to keep the old behviour for unix or am I missing something.

Reply:
The OR (`|`) without another alternative after means it will always find a zero-length match.

I think this is redundent since `\/?` matches `\` zero or one time. 

I'll add some more tests without a leading root to ensure this is the case.